### PR TITLE
Add extended sequential and repeat

### PR DESCRIPTION
**Pull request by belldandyxtq**
This PR add repeat functionality to torch.nn.sequential.
The repeat behaves the same way as the one in chainer.

The tests cases are copied from chainer and modified to pytorch.

### CODE REVIEW

File: pytorch_pfn_extras/nn/modules/extended_sequential.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+class ExtendedSequential(torch.nn.Sequential):
+    """Sequential module with extended features from chainer.
+
+    """
+    def _copy_model(self, mode):
+        if mode == 'init':
+            return _reset_parameters(copy.deepcopy(self))
+        elif mode == 'copy':
+            return copy.deepcopy(self)
+        else:
```
Comment (emcastillo) [inline pytorch_pfn_extras/nn/modules/extended_sequential.py:29]: nit:
How about checking the values and raise an error for unknown modes?

Comment (belldandyxtq) [reply pytorch_pfn_extras/nn/modules/extended_sequential.py:29]: I think the mode is checked in https://github.com/pfnet/pytorch-pfn-extras/blob/b5bc392feedd0f5c982180d67f11a7d14652f00d/pytorch_pfn_extras/nn/modules/extended_sequential.py#L60

Comment (emcastillo) [reply pytorch_pfn_extras/nn/modules/extended_sequential.py:29]: Fair enough!

**Comment by emcastillo** [comment]
Please, fix flake8 errors

**Comment by belldandyxtq** [comment]
Sorry, I just added more coverage to the test, can you please review it again? @emcastillo

**Pull request merged**